torch_xla/csrc/xla_lower_util.cpp

```diff
@@ -820,4 +820,45 @@ std::vector<xla::XlaOp> BuildAmpUpdateScale(const xla::XlaOp& current_scale,
   return results;
 }
 
+std::vector<xla::XlaOp> BuildSgdOptimizerStep(
+    const xla::XlaOp& found_inf, const xla::XlaOp& step,
+    const xla::XlaOp& param, const xla::XlaOp& d_p, const xla::XlaOp& buf,
+    const xla::XlaOp& weight_decay, const xla::XlaOp& momentum,
+    const xla::XlaOp& lr, const xla::XlaOp& dampening, bool use_weight_decay,
+    bool use_momentum, bool use_nesterov) {
+  /* XLA version of the SGD algorithm
+   * https://github.com/pytorch/pytorch/blob/master/torch/optim/_functional.py#L162-L180
+   */
+  auto dtype = XlaHelpers::ShapeOfXlaOp(param).element_type();
```

Comment:
We usually use `dtype` to refer to `at::ScalarType` which is a type from pytorch core. What you have here is a `XlaPrimitiveType` so you can just call it `type` or `para_type`.